akka-remote-tests/src/multi-jvm/scala/akka/remote/artery/MaxThroughputSpec.scala

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit.NANOSECONDS
+import scala.concurrent.duration._
+import akka.actor._
+import akka.remote.RemoteActorRefProvider
+import akka.remote.testconductor.RoleName
+import akka.remote.testkit.MultiNodeConfig
+import akka.remote.testkit.MultiNodeSpec
+import akka.remote.testkit.STMultiNodeSpec
+import akka.testkit._
+import com.typesafe.config.ConfigFactory
+
+object MaxThroughputSpec extends MultiNodeConfig {
+  val first = role("first")
+  val second = role("second")
+
+  commonConfig(debugConfig(on = false).withFallback(
+    ConfigFactory.parseString("""
+       MaxThroughputSpec.totalMessagesFactor = 1.0
+       akka {
+         loglevel = INFO
+         testconductor.barrier-timeout = 60s
+         actor {
+           provider = "akka.remote.RemoteActorRefProvider"
+           serialize-creators = false
+           serialize-messages = false
+         }
+         remote.artery.enabled = on
+       }
+       """)))
+
+  def aeronPort(roleName: RoleName): Int =
+    roleName match {
+      case `first`  ⇒ 20501 // TODO yeah, we should have support for dynamic port assignment
+      case `second` ⇒ 20502
+    }
+
+  nodeConfig(first) {
+    ConfigFactory.parseString(s"""
+      akka.remote.artery.port = ${aeronPort(first)}
+      """)
+  }
+
+  nodeConfig(second) {
+    ConfigFactory.parseString(s"""
+      akka.remote.artery.port = ${aeronPort(second)}
+      """)
+  }
+
+  case object Run
+  sealed trait Echo
+  final case object Start extends Echo
+  final case object End extends Echo
+  final case class EndResult(totalReceived: Long)
+  final case class FlowControl(burstStartTime: Long) extends Echo
+
+  def receiverProps(reporter: RateReporter, payloadSize: Int): Props =
+    Props(new Receiver(reporter, payloadSize))
+
+  class Receiver(reporter: RateReporter, payloadSize: Int) extends Actor {
+    var c = 0L
+
+    def receive = {
+      case Start ⇒
+        c = 0
+        sender() ! Start
```

Comment:
currently, you are better off caching the sender as it will point to a new `RemoteActorRef` instance every time, resulting in a lookup in the Association table for each send.